core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2204,11 +2209,22 @@ private void processTxFromBestChain(Transaction tx, boolean forceAddToPool) thro
             addWalletTransaction(Pool.SPENT, tx);
         }
 
-        // Kill txns in conflict with this tx
-        Set<Transaction> doubleSpendTxns = findDoubleSpendsAgainst(tx, pending);
-        if (!doubleSpendTxns.isEmpty()) {
-            // no need to addTransactionsDependingOn(doubleSpendTxns) because killTxns() already kills dependencies;
-            killTxns(doubleSpendTxns, tx);
+        if(!findDoubleSpendsAgainst(tx, transactions).isEmpty()) {
```

Comment:
The only time this could cause potential problems is if we get a transaction with no outputs to/from wallet which is not a double spend, any idea if that is possible?